tm2/pkg/p2p/transport.go

```diff
@@ -142,44 +139,35 @@
 	}
 
 	mt.cancelFn()
-
 	return mt.listener.Close()
 }
 
 // Listen starts an active process of listening for incoming connections [NON-BLOCKING]
-func (mt *MultiplexTransport) Listen(addr types.NetAddress) (rerr error) {
+func (mt *MultiplexTransport) Listen(addr types.NetAddress) error {
 	// Reserve a port, and start listening
 	ln, err := net.Listen("tcp", addr.DialString())
 	if err != nil {
 		return fmt.Errorf("unable to listen on address, %w", err)
 	}
 
-	defer func() {
-		if rerr != nil {
-			ln.Close()
-		}
-	}()
-
 	if addr.Port == 0 {
 		// net.Listen on port 0 means the kernel will auto-allocate a port
 		// - find out which one has been given to us.
 		tcpAddr, ok := ln.Addr().(*net.TCPAddr)
 		if !ok {
+			ln.Close()
```

Comment:
Are you sure `net.Listen` doesn't actually initialize a listener even despite the error?
I vaguely remember this being the case, this is why the close is captured in a defer